server/backend/check.ml

```diff
@@ -308,8 +309,17 @@ fi |} pkg pkg pkg (Server_configfile.platform_distribution conf)
         Printf.sprintf {|cd %s|} pkg;
         (* some projects only have a plain opam file, rename to pkg_name.opam is safe *)
         Printf.sprintf "if [ -f opam ]; then mv opam %s.opam; fi" pkg_name;
+
+        (* find all packages in the repo *)
+        Printf.sprintf "echo %s >> /tmp/packages-in-repo" pkg_name;
+        (* populate them from opam files *)
+        "for opam in *.opam; do echo ${opam%.opam} >> /tmp/packages-in-repo; done";
+        (* not all repos have opam files, populate them from dune-project *)
+        {|sed -n 's/(name \(.*\))/\1/p' < dune-project | sed 's/^[[:blank:]]*//;s/[[:blank:]]*$//' >> /tmp/packages-in-repo|};
```

Comment:
Isn't there a risk to catch more than only package name?

Reply:
Yes, it might catch the `(name cppo)` at the top level but I think that's always (?) benign?

Reply:
It should be the same as the one in the package, no? In this case, it is not a problem. @art-w was it your 2-cents on this?

Reply:
I would think so. In `cppo` it is `(name cppo)` along with 2 packages named `cppo` and `cppo_ocamlbuild` thus it doesn't create issues, because it is only used to create the opam files via `opam show`.

If this proves to be an issue on other packages, it can be fixed in a more comprehensive way.